File: src/main/java/com/yazan98/actor/annotations/ActorController.java
package com.yazan98.actor.annotations;

import com.yazan98.actor.ActorConsts;

import java.lang.annotation.*;

@Inherited
@Documented
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface ActorController {

    String name();
    String description() default ActorConsts.DEFAULT_CONTROLLER_DESCRIPTION;

}

File: src/main/java/com/yazan98/actor/annotations/ActorController.kt
package com.yazan98.actor.annotations

import java.lang.annotation.Inherited


@Inherited
@MustBeDocumented
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class ActorController(
        val name: String,
        val path: String,
        val description: String
)
